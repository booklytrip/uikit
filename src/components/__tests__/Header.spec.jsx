import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Header, SubHeader } from '../Header';

describe('Header', () => {
    it('renders header without crashing', () => {
        mount(<Header />);
    });

    it('renders sub-header without crashing', () => {
        mount(<SubHeader />);
    });

    describe('Snapshots', () => {
        it('renders header of different type', () => {
            const rendered = renderer.create(
                <Header type="h2">My Header</Header>
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders header with an icon', () => {
            const rendered = renderer.create(
                <Header icon>
                    My Header
                </Header>
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders centered header', () => {
            const rendered = renderer.create(
                <Header centered>
                    My Header
                </Header>
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders dividing header', () => {
            const rendered = renderer.create(
                <Header dividing>
                    My Header
                </Header>
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
