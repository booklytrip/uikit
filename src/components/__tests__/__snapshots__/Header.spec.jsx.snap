exports[`Header Snapshots renders centered header 1`] = `
<h3
  className="ui header centered"
  style={undefined}>
  My Header
</h3>
`;

exports[`Header Snapshots renders dividing header 1`] = `
<h3
  className="ui header dividing"
  style={undefined}>
  My Header
</h3>
`;

exports[`Header Snapshots renders header of different type 1`] = `
<h2
  className="ui header"
  style={undefined}>
  My Header
</h2>
`;

exports[`Header Snapshots renders header with an icon 1`] = `
<h3
  className="ui header icon"
  style={undefined}>
  My Header
</h3>
`;
